src/lib.rs

```diff
@@ -54,7 +54,7 @@ pub use error::Error;
 /// A hashing engine which bytes can be serialized into. It is expected
 /// to implement the `io::Write` trait, but to never return errors under
 /// any conditions.
-pub trait HashEngine: Clone + io::Write {
```

Comment:
I personally like `HashEngine` better since it at least tells you *something* about it's function.